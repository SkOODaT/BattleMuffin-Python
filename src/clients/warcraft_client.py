import requests
from authlib.integrations.requests_client import OAuth2Session
from uplink import Consumer, Query, get, params, returns

from config.client_config import ClientConfiguration
from config.region_config import Locale, Region


class WarcraftClient(Consumer):
    def __init__(self, client_id: str, client_secret: str, region: Region = Region.us, locale: Locale = Locale.en_US):
        self.client_config = ClientConfiguration(client_id, client_secret, region, locale)

        oath_response = requests.get(f"{self.client_config.oauth_host}/oauth/.well-known/openid-configuration").json()
        token_endpoint = oath_response['token_endpoint']
        session = OAuth2Session(client_id, client_secret, scope="openid", token_endpoint=token_endpoint, grant_type="client_credentials")
        session.fetch_token()
        super().__init__(base_url=self.client_config.host, client=session)
        self._inject(Query("locale").with_value(self.client_config.locale.name))

    @returns.json
    @params({"namespace": "static-us"})
    @get("/data/wow/achievement-category/index")
    def get_achievement_category_index(self):
        """Returns an index of achievement categories."""
