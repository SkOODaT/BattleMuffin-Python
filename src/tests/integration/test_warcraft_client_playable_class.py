import os
from clients.warcraft_client import WarcraftClient


def test_get_playable_classes_index(snapshot):
    client = WarcraftClient(os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"))
    response = client.get_playable_classes_index()
    assert response == snapshot


def test_get_playable_class(snapshot):
    client = WarcraftClient(os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"))
    response = client.get_playable_class(7)
    assert response == snapshot


def test_get_playable_class_media(snapshot):
    client = WarcraftClient(os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"))
    response = client.get_playable_class_media(7)
    assert response == snapshot


def test_get_pvp_talent_slots(snapshot):
    client = WarcraftClient(os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"))
    response = client.get_pvp_talent_slots(7)
    assert response == snapshot
