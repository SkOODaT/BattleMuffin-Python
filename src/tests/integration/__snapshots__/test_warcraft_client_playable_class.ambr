# name: test_get_playable_class
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/playable-class/7?namespace=static-8.3.0_32861-us',
      },
    },
    'gender_name': <class 'dict'> {
      'female': 'Shaman',
      'male': 'Shaman',
    },
    'id': 7,
    'media': <class 'dict'> {
      'id': 7,
      'key': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/media/playable-class/7?namespace=static-8.3.0_32861-us',
      },
    },
    'name': 'Shaman',
    'power_type': <class 'dict'> {
      'id': 0,
      'key': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/power-type/0?namespace=static-8.3.0_32861-us',
      },
      'name': 'Mana',
    },
    'pvp_talent_slots': <class 'dict'> {
      'href': 'https://us.api.blizzard.com/data/wow/playable-class/7/pvp-talent-slots?namespace=static-8.3.0_32861-us',
    },
    'specializations': <class 'list'> [
      <class 'dict'> {
        'id': 262,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/262?namespace=static-8.3.0_32861-us',
        },
        'name': 'Elemental',
      },
      <class 'dict'> {
        'id': 263,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/263?namespace=static-8.3.0_32861-us',
        },
        'name': 'Enhancement',
      },
      <class 'dict'> {
        'id': 264,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/264?namespace=static-8.3.0_32861-us',
        },
        'name': 'Restoration',
      },
    ],
  }
---
# name: test_get_playable_class_media
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/media/playable-class/7?namespace=static-8.3.0_32861-us',
      },
    },
    'assets': <class 'list'> [
      <class 'dict'> {
        'key': 'icon',
        'value': 'https://render-us.worldofwarcraft.com/icons/56/classicon_shaman.jpg',
      },
    ],
    'id': 7,
  }
---
# name: test_get_playable_classes_index
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/playable-class/?namespace=static-8.3.0_32861-us',
      },
    },
    'classes': <class 'list'> [
      <class 'dict'> {
        'id': 1,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-class/1?namespace=static-8.3.0_32861-us',
        },
        'name': 'Warrior',
      },
      <class 'dict'> {
        'id': 2,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-class/2?namespace=static-8.3.0_32861-us',
        },
        'name': 'Paladin',
      },
      <class 'dict'> {
        'id': 3,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-class/3?namespace=static-8.3.0_32861-us',
        },
        'name': 'Hunter',
      },
      <class 'dict'> {
        'id': 4,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-class/4?namespace=static-8.3.0_32861-us',
        },
        'name': 'Rogue',
      },
      <class 'dict'> {
        'id': 5,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-class/5?namespace=static-8.3.0_32861-us',
        },
        'name': 'Priest',
      },
      <class 'dict'> {
        'id': 6,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-class/6?namespace=static-8.3.0_32861-us',
        },
        'name': 'Death Knight',
      },
      <class 'dict'> {
        'id': 7,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-class/7?namespace=static-8.3.0_32861-us',
        },
        'name': 'Shaman',
      },
      <class 'dict'> {
        'id': 8,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-class/8?namespace=static-8.3.0_32861-us',
        },
        'name': 'Mage',
      },
      <class 'dict'> {
        'id': 9,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-class/9?namespace=static-8.3.0_32861-us',
        },
        'name': 'Warlock',
      },
      <class 'dict'> {
        'id': 10,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-class/10?namespace=static-8.3.0_32861-us',
        },
        'name': 'Monk',
      },
      <class 'dict'> {
        'id': 11,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-class/11?namespace=static-8.3.0_32861-us',
        },
        'name': 'Druid',
      },
      <class 'dict'> {
        'id': 12,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-class/12?namespace=static-8.3.0_32861-us',
        },
        'name': 'Demon Hunter',
      },
    ],
  }
---
# name: test_get_pvp_talent_slots
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/playable-class/7/pvp-talent-slots?namespace=static-8.3.0_32861-us',
      },
    },
    'talent_slots': <class 'list'> [
      <class 'dict'> {
        'slot_number': 1,
        'unlock_player_level': 20,
      },
      <class 'dict'> {
        'slot_number': 2,
        'unlock_player_level': 40,
      },
      <class 'dict'> {
        'slot_number': 3,
        'unlock_player_level': 70,
      },
      <class 'dict'> {
        'slot_number': 4,
        'unlock_player_level': 110,
      },
    ],
  }
---
